core/ingest/src/main/java/mil/nga/giat/geowave/core/ingest/spark/SparkIngestIngestDriver.java

```diff
@@ -0,0 +1,717 @@
+import mil.nga.giat.geowave.core.store.operations.remote.options.DataStorePluginOptions;
+import mil.nga.giat.geowave.core.store.operations.remote.options.IndexLoader;
+import mil.nga.giat.geowave.core.store.operations.remote.options.IndexPluginOptions;
+import mil.nga.giat.geowave.core.store.operations.remote.options.StoreLoader;
+import mil.nga.giat.geowave.core.store.operations.remote.options.VisibilityOptions;
+import mil.nga.giat.geowave.mapreduce.operations.ConfigHDFSCommand;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.beust.jcommander.ParameterException;
+import com.google.common.base.Function;
+import com.google.common.collect.Lists;
+import com.upplication.s3fs.S3FileSystem;
+import com.upplication.s3fs.S3FileSystemProvider;
+import com.upplication.s3fs.S3Path;
+
+public class SparkIngestIngestDriver implements
+		Serializable
+{
+
+	private final static Logger LOGGER = LoggerFactory.getLogger(SparkIngestIngestDriver.class);
+
+	public SparkIngestIngestDriver() {
+
+	}
+
+	public boolean runOperation(
+			File configFile,
+			LocalInputCommandLineOptions localInput, String inputStoreName,
+			String indexList, VisibilityOptions ingestOptions,
+			SparkCommandLineOptions sparkOptions, String basePath)
+			throws IOException {
+
+		final Properties configProperties = ConfigOptions.loadProperties(
+				configFile,
+				null);
+		
+		JavaSparkContext jsc = null;
+		SparkSession session = null;
+		int numExecutors ;
+		int numCores ;
+		int numPartitions;
+		
+		if (jsc == null) {
+			String jar = "";
+			try {
+				jar = SparkIngestIngestDriver.class.getProtectionDomain()
+						.getCodeSource().getLocation().toURI().getPath();
+			} catch (final URISyntaxException e) {
+				LOGGER.error(
+						"Unable to set jar location in spark configuration", e);
+			}
+
+			session = SparkSession.builder().appName(sparkOptions.getAppName())
+					.master(sparkOptions.getMaster())
+					.config("spark.driver.host", sparkOptions.getHost())
+					.config("spark.jars", jar)
+					.getOrCreate();
+
+			jsc = new JavaSparkContext(session.sparkContext());
+		}
+
+		// If input path is S3
+		if (basePath.startsWith("s3://")) {
+			
+			String s3EndpointUrl = ConfigAWSCommand.getS3Url(configProperties);
+			
+			Path inpuPath = (S3Path) setUpS3Filesystem(configProperties, basePath,s3EndpointUrl);
+			if (inpuPath == null || !inpuPath.isAbsolute()) {
+				LOGGER.error("Error in accessing S3 Input path " + basePath);
+				close(jsc,session);
+				return false;
+			}
+			
+			List<Path> inputFileList = new ArrayList<Path>();
+			Files.walkFileTree(inpuPath, new SimpleFileVisitor<Path>() {
+			    
+			    @Override
+			    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+			      inputFileList.add(file);
+			      return FileVisitResult.CONTINUE;
+			    }
+			  });
+		
+			int numInputFiles  = inputFileList.size();
+	
+			if (sparkOptions.getNumExecutors() == 1) {
```

Comment:
shouldn't we use something like -1 as the default value and check if its less than 1 here (same with numCores)?

Seems like someone could actually want 1 to be a valid value